cmreactor/cmreactor.sh

```
@@ -0,0 +1,479 @@
+    gitp config -f "${metaFile}" "cmreactor.meta.${module}.branch" "$(git --no-pager -C "${modulePath}" rev-parse --abbrev-ref HEAD)"
+    gitp config -f "${metaFile}" "cmreactor.meta.${module}.commit" "$(git --no-pager -C "${modulePath}" rev-parse HEAD)"
+    gitp config -f "${metaFile}" "cmreactor.meta.${module}.oneline" "$(git --no-pager -C "${modulePath}" show -s --oneline)"
+}
+
+recordReactorGitMeta() {
+    local metaFile
+    metaFile="$(pathInProject "cmreactor-meta.gitconfig")"
+
+    gitp config -f "${metaFile}" "cmreactor.reactor-meta.origin" "$(gitc remote get-url origin)"
+    gitp config -f "${metaFile}" "cmreactor.reactor-meta.branch" "$(gitc rev-parse --abbrev-ref HEAD)"
+    gitp config -f "${metaFile}" "cmreactor.reactor-meta.commit" "$(gitc rev-parse HEAD)"
+    gitp config -f "${metaFile}" "cmreactor.reactor-meta.oneline" "$(gitc show -s --oneline)"
+}
+
+syncMavenRepositoryFolders() {
+    find "${projectPath}" -name maven_repository -type d -mindepth 2 \
+        -exec rsync -a '{}'/ "${projectPath}/maven_repository/" \;
+}
+### END CMREACTOR FUNCTIONS ###
+
+if [[ "${projectPath}" == "/" ]]; then
+  echo "$(basename "${BASH_SOURCE[0]}") prevented root path removal" >&2
+  exit 1
+fi
+
+[[ -d "${projectPath}" ]] && rm -rf "${projectPath}"
+mkdir -p "${projectPath}"
+
+declare -a moduleNames
+moduleNames=($(readModulesFromConfig "${configFile}"))
+
+reactorPomInit "${moduleNames[@]}"
+recordReactorGitMeta
+
+if [[ "${moduleNames[*]}" != "" ]]; then
+    tempBase="$(pathInSource target/cmreactor-tmp)"
+    mkdir -p "${tempBase}"
+    for moduleName in "${moduleNames[@]}"; do
+        moduleType="$(readModuleConfigValue "${configFile}" "${moduleName}" "type")"
+        case "$moduleType" in
+        zip)    handleZipModule "${configFile}" "${moduleName}" "${tempBase}";;
+        git)    handleGitModule "${configFile}" "${moduleName}" "${tempBase}";;
+        *)      echo "Unsupported module type $moduleType for module $moduleName" >&2
```

Should this trigger a failure rather than continuing execution?

---

Yes, good idea.